File: Harjutus6/settingudmobiilis/app/src/main/java/com/example/settingutememorize/MainActivity.java
package com.example.settingutememorize;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.TextureView;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.TextView;

import com.google.android.material.switchmaterial.SwitchMaterial;

public class MainActivity extends AppCompatActivity {

    private View parentview;
    private SwitchMaterial themeSwitch;
    private TextView themeTV, titleTV;

    private userSettings settings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        settings = (userSettings) getApplication();
        initWidgets();
        loadSharedPreferences();
        initSwitchListener();
        
    }

    private void initSwitchListener() {
        themeSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean checked) {
                if(checked)
                    settings.setCustomTheme(userSettings.DARK_THEME);
                else
                    settings.setCustomTheme(userSettings.LIGHT_THEME);

                SharedPreferences.Editor editor = getSharedPreferences(userSettings.PREFERENCES, MODE_PRIVATE).edit();
                editor.putString(userSettings.CUSTOM_THEME,settings.getCustomTheme());
                editor.apply();

                updateView();
            }

        });
    }

    private void loadSharedPreferences() {
        SharedPreferences sharedPreferences = getSharedPreferences(userSettings.PREFERENCES,MODE_PRIVATE);
        String theme = sharedPreferences.getString(userSettings.CUSTOM_THEME,userSettings.LIGHT_THEME);
        settings.setCustomTheme(theme);
        updateView();
    }

    private void initWidgets() {
        themeTV = findViewById(R.id.themeTV);
        titleTV = findViewById(R.id.titleTV);
        themeSwitch = findViewById(R.id.themeSwitch);
        parentview = findViewById(R.id.parentView);
    }

    private void updateView() {
        final int black = ContextCompat.getColor(this, R.color.black);
        final int white = ContextCompat.getColor(this, R.color.white);

        if(settings.getCustomTheme().equals(userSettings.DARK_THEME)) {
            titleTV.setTextColor(white);
            themeTV.setTextColor(white);
            themeTV.setText("Dark");
            parentview.setBackgroundColor(black);
            themeSwitch.setChecked(true);
        }else {
            titleTV.setTextColor(black);
            themeTV.setTextColor(black);
            themeTV.setText("Light");
            parentview.setBackgroundColor(white);
            themeSwitch.setChecked(false);

        }
    }

}

File: Harjutus4kooskaameraga/app/src/main/java/codes/drinky/testapp/MainActivity.kt
package codes.drinky.testapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private lateinit var openCounter: Button
    private lateinit var openJokes: Button
    private lateinit var openAccel: Button
    private lateinit var openCamera: Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        openCounter = findViewById<Button>(R.id.openCounter)
        openJokes = findViewById<Button>(R.id.openJokes)
        openAccel = findViewById<Button>(R.id.openAccelerometer)
        openCamera = findViewById<Button>(R.id.openCamera)
        setListeners()
    }

    private fun setListeners() {
        openCounter.setOnClickListener { openCounterActivity() }
        openJokes.setOnClickListener { openJokesActivity() }
        openAccel.setOnClickListener { openAcceleratorActivity() }
        openCamera.setOnClickListener { openCameraActivity() }

    }

    private fun openCounterActivity() {
        val intent = Intent(this, CounterActivity::class.java)
        startActivity(intent)
    }

    private fun openAcceleratorActivity() {
        val intent = Intent(this, AcceleratorActivity::class.java)
        startActivity(intent)
    }

    private fun openJokesActivity() {
        val intent = Intent(this, JokesActivity::class.java)
        startActivity(intent)
    }

    private fun openCameraActivity() {
        val intent = Intent(this, CameraActivity::class.java)
        startActivity(intent)
    }


}